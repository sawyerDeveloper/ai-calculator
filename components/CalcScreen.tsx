import { StyleSheet, Text, View } from 'react-native';
import { AiThinker } from './AiThinker';
import { useState } from 'react';

interface CalcScreenProps {
  value: [number | string];
  loading: boolean;
}
export function CalcScreen({ value, loading }: CalcScreenProps) {
  const [showAnswer, setShowAnswer] = useState(false);
  return (
    <View style={styles.container}>
      <Text>{value.map((digit) => digit)}</Text>
      {loading && !showAnswer && (
        <AiThinker aiCallback={() => setShowAnswer(true)} aiModel={value} />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 30,
    backgroundColor: 'white',
    borderColor: 'black',
    borderWidth: 1,
    justifyContent: 'center',
    fontSize: 30,
    paddingLeft: 10,
  },
});
