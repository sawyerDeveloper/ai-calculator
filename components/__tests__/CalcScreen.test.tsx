import { render } from '@testing-library/react-native';
import { CalcScreen } from '../CalcScreen';

describe('CalcScreen', () => {
  it('renders correctly', () => {
    const calcScreen = render(
      <CalcScreen value={[1, '+', 1, '=', 2]} loading={false} />
    )
    expect(calcScreen.toJSON()).toMatchSnapshot();
    expect(calcScreen.findByText('1+1=2'))
  });
});
