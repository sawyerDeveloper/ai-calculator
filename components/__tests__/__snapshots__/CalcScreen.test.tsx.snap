// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcScreen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderColor": "black",
      "borderWidth": 1.5,
      "height": 60,
      "justifyContent": "center",
      "paddingLeft": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 30,
      }
    }
  >
    1
    +
    1
    =
    2
  </Text>
</View>
`;
